apps/server/src/modules/tool/uc/lti11.uc.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable, InternalServerErrorException } from '@nestjs/common';
+import { CustomLtiProperty, LtiToolDO } from '@shared/domain/domainobject/ltitool.do';
+import { LtiToolRepo } from '@shared/repo';
+import { ICurrentUser, LtiPrivacyPermission, RoleName } from '@shared/domain';
+import { UserService } from '@src/modules/user/service/user.service';
+import { UserDto } from '@src/modules/user/uc/dto/user.dto';
+import OAuth, { Authorization, RequestOptions } from 'oauth-1.0a';
+import { Lti11Service } from '../service/lti11.service';
+import { Lti11PayloadDto } from './dto/lti11-payload.dto';
+import { LtiRole } from '../interface/lti-role.enum';
+import { LtiRoleMapper } from '../mapper/lti-role.mapper';
+
+@Injectable()
+export class Lti11Uc {
+	constructor(
+		private readonly lti11Service: Lti11Service,
+		private readonly ltiRoleMapper: LtiRoleMapper,
+		private readonly ltiToolRepo: LtiToolRepo,
+		private readonly userService: UserService
+	) {}
+
+	async getLaunchParameters(currentUser: ICurrentUser, toolId: string, courseId: string): Promise<Authorization> {
```

Comment:
think about splitting this function up in some private function

Reply:
done